lib/rpc/responses/spec.ex

```diff
@@ -0,0 +1,11 @@
+defmodule Soroban.RPC.Response.Spec do
+  @moduledoc """
+  Defines RPC response types constructions.
+  """
+
+  @type attrs :: map() | struct() | Keyword.t() | list() | String.t()
+  @type error :: {:error, Keyword.t()}
+  @type resource :: struct()
+
+  @callback new(any()) :: struct() | error()
```

Comment:
I think the `new/1` function will never return an error... 🤔 these response types will be created from a map result of the HTTP request